core/runtime/src/main/java/io/quarkus/runtime/configuration/SystemOnlySourcesConfigBuilder.java

```diff
@@ -0,0 +1,17 @@
+package io.quarkus.runtime.configuration;
+
+import io.smallrye.config.SmallRyeConfigBuilder;
+
+public class SystemOnlySourcesConfigBuilder implements ConfigBuilder {
+    @Override
+    public SmallRyeConfigBuilder configBuilder(final SmallRyeConfigBuilder builder) {
+        // TODO - This still discovers sources in the classpath. Should we also remove them?
```

Comment:
Config sources on the classpath (where they are defaults) is fine.. those would also be present at build time.

My thinking was always that constraining to system resources was solely/only a runtime concern.

Reply:
Yes, nothing here changes what happens at build time.

This will remove sources that contain `application.properties` and the `.env` file.

Reply:
(can we remove the todo?)

Reply:
Yes. Done!